fe/fe-core/src/main/java/org/apache/doris/datasource/property/fileformat/FileFormatProperties.java

```diff
@@ -0,0 +1,116 @@
+    protected TFileFormatType fileFormatType;
+
+    protected TFileCompressType compressionType;
+
+    public FileFormatProperties(TFileFormatType fileFormatType) {
+        this.fileFormatType = fileFormatType;
+    }
+
+    /**
+     *
+     * @param formatProperties
+     * @return properties needed by Doris
+     * @throws AnalysisException
+     */
+
+    /**
+     * Analyze user properties
+     * @param formatProperties properties specified by user
+     * @param isRemoveOriginProperty if this param is set to true, then this method would remove the origin property
+     * @throws AnalysisException
+     */
+    public abstract void analyzeFileFormatProperties(
+            Map<String, String> formatProperties, boolean isRemoveOriginProperty)
+            throws AnalysisException;
+
+    public abstract PFetchTableSchemaRequest toPFetchTableSchemaRequest();
+
+    public abstract TResultFileSinkOptions toTResultFileSinkOptions();
+
+    public abstract TFileAttributes toTFileAttributes();
+
+    public static FileFormatProperties createFileFormatProperties(String formatString) {
+        switch (formatString) {
+            case FileFormatBaseProperties.FORMAT_CSV:
+                return new CsvFileFormatProperties(TFileFormatType.FORMAT_CSV_PLAIN);
+            case FileFormatBaseProperties.FORMAT_HIVE_TEXT:
+                return new CsvFileFormatProperties(TFileFormatType.FORMAT_CSV_PLAIN,
+                        CsvProperties.DEFAULT_HIVE_TEXT_COLUMN_SEPARATOR,
+                        TTextSerdeType.HIVE_TEXT_SERDE);
+            case FileFormatBaseProperties.FORMAT_CSV_WITH_NAMES:
+                return new CsvFileFormatProperties(TFileFormatType.FORMAT_CSV_PLAIN,
+                        FileFormatBaseProperties.FORMAT_CSV_WITH_NAMES);
+            case FileFormatBaseProperties.FORMAT_CSV_WITH_NAMES_AND_TYPES:
+                return new CsvFileFormatProperties(TFileFormatType.FORMAT_CSV_PLAIN,
+                        FileFormatBaseProperties.FORMAT_CSV_WITH_NAMES_AND_TYPES);
+            case FileFormatBaseProperties.FORMAT_PARQUET:
+                return new ParquetFileFormatProperties(TFileFormatType.FORMAT_PARQUET);
+            case FileFormatBaseProperties.FORMAT_ORC:
+                return new OrcFileFormatProperties(TFileFormatType.FORMAT_ORC);
+            case FileFormatBaseProperties.FORMAT_JSON:
+                return new JsonFileFormatProperties(TFileFormatType.FORMAT_JSON);
+            case FileFormatBaseProperties.FORMAT_AVRO:
+                return new AvroFileFormatProperties(TFileFormatType.FORMAT_AVRO);
+            case FileFormatBaseProperties.FORMAT_WAL:
+                return new WalFileFormatProperties(TFileFormatType.FORMAT_WAL);
+            default:
+                throw new AnalysisException("format:" + formatString + " is not supported.");
+        }
+    }
+
+    public static FileFormatProperties createFileFormatProperties(Map<String, String> formatProperties)
```

Comment:
Sometimes we may use the suffix of a file to guess its format.
For example, the file path is /path/to/1.parquet, so we know that this is a parquet format, even if user does not specify the "file_format" property.
We should handle this.

1. create a tool method to infer the file format from path.
2. change the signature of this `createFileFormatProperties` to `createFileFormatProperties(TFileFormatType fileFormat, properties)`

And I think without the suffix and user specified "file_format" property, we should provide a "default" format file, eg, csv. But it depends because I am not sure what the previous logic is